internal/test/smoke/system_endpoints_test.go

```diff
@@ -87,20 +95,60 @@ func (s *SuiteSystemEndpoints) TestHostConfExecute() {
 	t := s.T()
 	s.As(RBACSuperAdmin)
 	s.prepareDomainIpa(t)
-
-	t.Log("Calling SystemHostConfWithResponse")
 	domainType := public.RhelIdm
 	s.As(XRHIDSystem, RBACNoPermis)
-	res, err := s.HostConfWithResponse(
-		s.domain.RhelIdm.Servers[0].SubscriptionManagerId.String(),
-		"client."+s.domain.DomainName,
-		builder_api.NewHostConf().
-			WithDomainName(pointy.String(s.domain.DomainName)).
-			WithDomainType(&domainType).
-			Build())
-	require.NoError(t, err)
-	require.NotNil(t, res)
-	require.Equal(t, http.StatusOK, res.StatusCode)
+
+	t.Run("HostConf success", func(t *testing.T) {
```

Comment:
I suggest to split this in two different test methods: `TestHostConfExecuteSuccess` and `TestHostConfExecuteFailure`, and move the common code to a private method. Only for consistency; I am not familiar using directly the `t.Run` method, but happy to know more about it.

Every `Test...` method would have a fresh and clean mock instance to add the expectations for SendTrackEvent.